README.md

```diff
@@ -354,4 +354,36 @@ The Azure DevOps pipeline will run automatically every day at 8am to refresh the
 
 When the timestamp is surpassed by the current time, the `time_rotating` resource will be refreshed and the `azuread_application_password` token will be updated to a newly generated value.
 
-This value will then be used to update the service connection in Azure DevOps.
\ No newline at end of file
+This value will then be used to update the service connection in Azure DevOps.
+
+## Creating and assigning custom roles
+
+### Creating a custom role 
+
+To create a custom role you need to update the [locals.tf](./modules/enterprise/locals.tf) file in the enterprise module and update the `custom_roles` local value with the name of the role, the description, the scope and the actions allowed.
+
+```terraform
+  custom_roles = {
+    "Application Gateway Backend Health Reader" = {
+      description = "View backend health on the Application Gateway"
+      scope       = "/providers/Microsoft.Management/managementGroups/HMCTS"
+      actions     = ["Microsoft.Network/applicationGateways/backendhealth/action"]
+    },
+  }
+```
+
+### Assigning custom roles
+
+To assign a Custom Role, the role itself needs to be created in this repo first using the instructions above.
+
+Once the role has been created we can assign it by updating the [locals.tf](./modules/subscription/locals.tf) file in the subscription module. You will need to add the name of the custom role, the ID of the principal being assigned to the role and the scope of the permission.
```

Comment:
what about assigning it at management groups? permissions should be able to be applied there too? (Can't see any changes in this PR relating to MG permissions)

Reply:
Oo yeah ok, I'll sort that